test/intl402/DateTimeFormat/prototype/formatToParts/related-year-zh.js

```diff
@@ -12,18 +13,25 @@ locale: [zh-u-ca-chinese]
 function verifyFormatParts(actual, expected, message) {
   assert.sameValue(Array.isArray(expected), true, `${message}: expected is Array`);
   assert.sameValue(Array.isArray(actual), true, `${message}: actual is Array`);
-  assert.sameValue(actual.length, expected.length, `${message}: length`);
+  if (actual.length !== expected.length)
+    return false;
 
   for (let i = 0; i < actual.length; ++i) {
     assert.sameValue(actual[i].type, expected[i].type, `${message}: parts[${i}].type`);
     assert.sameValue(actual[i].value, expected[i].value, `${message}: parts[${i}].value`);
   }
+  return true;
 }
 
 const df = new Intl.DateTimeFormat("zh-u-ca-chinese", {year: "numeric"});
 const date = new Date(2019, 5, 1);
-verifyFormatParts(df.formatToParts(date), [
+const expected1 = [
   {type: "relatedYear", value: "2019"},
   {type: "yearName", value: "己亥"},
   {type: "literal", value: "年"},
-]);
+];
+const expected2 = [
+  {type: "yearName", value: "己亥"},
+  {type: "literal", value: "年"},
+];
+assert.sameValue(verifyFormatParts(df.formatToParts(date), expected1) || verifyFormatParts(df.formatToParts(date), expected2), true);
```

Comment:
This || breaks a bit of the test determinism and we can try to move this branching to a lower layer here. The test seems correct, but let's see if we can reduce this.

Reply:
Fixed! Thanks